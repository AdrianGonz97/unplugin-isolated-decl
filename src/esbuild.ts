import { IsolatedDecl } from './index'

export default IsolatedDecl.esbuild as typeof IsolatedDecl.esbuild
