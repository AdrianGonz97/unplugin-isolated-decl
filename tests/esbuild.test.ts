import { readdir, readFile } from 'node:fs/promises'
import path from 'node:path'
import { build } from 'esbuild'
import { describe, expect, test } from 'vitest'
import { dependencies } from '../package.json'
import UnpluginIsolatedDecl from '../src/esbuild'

describe('esbuild', () => {
  const input = path.resolve(__dirname, 'fixtures/main.ts')

  test('write mode', async () => {
    const dist = path.resolve(__dirname, 'temp')
    await build({
      entryPoints: [input],
      plugins: [UnpluginIsolatedDecl({ extraOutdir: 'temp' })],
      logLevel: 'silent',
      bundle: true,
      external: Object.keys(dependencies),
      platform: 'node',
      outdir: dist,
      format: 'esm',
    })

    const outDir = path.resolve(dist, 'temp')
    await expect(
      Promise.all(
        (await readdir(outDir))
          .sort()
          .map((file) => readFile(path.resolve(outDir, file), 'utf8')),
      ),
    ).resolves.toMatchSnapshot()
  })

  test('generate mode', async () => {
    const { outputFiles } = await build({
      entryPoints: [input],
      plugins: [UnpluginIsolatedDecl({ extraOutdir: 'temp' })],
      logLevel: 'silent',
      bundle: true,
      external: Object.keys(dependencies),
      platform: 'node',
      write: false,
      format: 'esm',
    })

    expect(
      outputFiles.map((file) => `// ${file.path}\n${file.text}`),
    ).toMatchSnapshot()
  })
})
