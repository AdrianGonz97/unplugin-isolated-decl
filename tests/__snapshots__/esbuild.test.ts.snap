// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`esbuild > generate mode 1`] = `
[
  "// <stdout>
// tests/fixtures/main.ts
function hello(s) {
  return "hello" + s;
}
var num = 1;
export {
  hello,
  num
};
",
  "// main.d.ts
import { type Num } from "./types";
export type Str = string;
export declare function hello(s: Str): Str;
export declare let num: Num;
",
  "// types.d.ts
export type Num = number;
",
]
`;

exports[`esbuild > write mode 1`] = `
[
  "import { type Num } from "./types";
export type Str = string;
export declare function hello(s: Str): Str;
export declare let num: Num;
",
  "// tests/fixtures/main.ts
function hello(s) {
  return "hello" + s;
}
var num = 1;
export {
  hello,
  num
};
",
  "export type Num = number;
",
]
`;
